components/iam/pkg/oidc/service_test.go

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package oidc
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/oauth2"
+)
+
+func TestGetStartParams(t *testing.T) {
+	const (
+		issuerG  = "https://accounts.google.com"
+		clientID = "client-id-123"
+	)
+	service := NewOIDCService()
+	config := &OIDCClientConfig{
+		ID:         "google-1",
+		Issuer:     issuerG,
+		OIDCConfig: &oidc.Config{},
+		OAuth2Config: &oauth2.Config{
+			ClientID: clientID,
+			Endpoint: oauth2.Endpoint{
+				AuthURL: issuerG + "/o/oauth2/v2/auth",
+			},
+		},
+	}
+
+	params, err := service.GetStartParams(config)
+
+	require.NoError(t, err)
+	require.NotNil(t, params.Nonce)
+	require.NotNil(t, params.State)
+
+	// AuthCodeURL example:
+	// https://accounts.google.com/o/oauth2/v2/auth
+	// ?client_id=client-id-123
+	// &nonce=UFTMxxUtc5jVZbp2a2R9XEoRwpfzs-04FcmVQ-HdCsw
+	// &response_type=code
+	// &state=Q4XzRcdo4jtOYeRbF17T9LHHwX-4HacT1_5pZH8mXLI
+	require.NotNil(t, params.AuthCodeURL)
+	require.Contains(t, params.AuthCodeURL, issuerG)
+	require.Contains(t, params.AuthCodeURL, clientID)
+	require.Contains(t, params.AuthCodeURL, params.Nonce)
+	require.Contains(t, params.AuthCodeURL, params.State)
+}
+
+func TestGetClientConfigFromRequest(t *testing.T) {
+	issuer, err := setupFakeIdP(t)
+	require.NoError(t, err)
+
+	testCases := []struct {
+		Location      string
+		ExpectedError bool
```

Comment:
Should be checking for HTTP Status codes, not errors as that's the interface on which we communicate

Reply:
At second glance, that's a test to check the request params we are to handle. That's not a response,